src/integrations/windowsIntegrationManager.ts

```diff
@@ -358,60 +388,60 @@ export default class WindowsIntegrationManager implements IntegrationManager {
     console.log(`kubeconfig integration for ${ distro } set to ${ state }`);
   }
 
-  /**
-   * List the registered WSL2 distributions.
-   */
-  protected get distros(): Promise<string[]> {
+  protected get nonBlacklistedDistros(): Promise<WSLDistro[]> {
     return (async() => {
-      const distros = (await this.captureCommand(
-        { encoding: 'utf16le' },
-        '--list', '--quiet'))
-        .split(/[\r\n]+/g)
-        .map(x => x.trim())
-        .filter(x => x);
-
-      if (distros.length < 1) {
-      // Return early if we find no distributions in this list; listing again
-      // with verbose will fail if there are no distributions.
-        return [];
-      }
-
-      const stdout = await this.captureCommand({ encoding: 'utf16le' }, '--list', '--verbose');
+      const wslOutput = await this.captureCommand({ encoding: 'utf16le' }, '--list', '--verbose');
```

Comment:
Totally missed this — this means that if no WSL distros are registered (i.e. before we registered our own), this will throw an exception (see the comment that used to be on line 374). The callers don't seem to be catching this (and I'd argue that's kind of a crappy API).

```pwsh
PS C:\Users\Admin> (Get-CimInstance Win32_OperatingSystem).Version
10.0.22000
PS C:\Users\Admin> wsl --list --quiet ; echo $LASTEXITCODE
0
PS C:\Users\Admin> wsl --list --verbose ; echo $LASTEXITCODE
Windows Subsystem for Linux has no installed distributions.
Distributions can be installed by visiting the Microsoft Store:
https://aka.ms/wslstore
-1
```

Reply:
Ok I wrapped this in a try-catch. I think that should take care of this case.